platform/mds/mds/src/main/java/org/motechproject/mds/repository/MotechDataRepository.java

```diff
@@ -268,4 +276,22 @@ private Query queryForFilters(Filters filters, QueryParams queryParams, Instance
 
         return query;
     }
+
+    private void preparePropertiesForGroupingTask(List<Property> properties, String groupingColumn,  InstanceSecurityRestriction restriction) {
```

Comment:
MotechDataRepository is for general purpose, you should not add specific code for tasks module in this class